src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -5,67 +5,302 @@
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" Mc vitty", "LA democh ", "josephinE-mc-vitty ", " sa-la-demoCh"), firstAndLastNameFailedMsg(" Mc vitty", "LA democh ", "josephinE-mc-vitty ", " sa-la-demoCh"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("  mC  ", "  La  ", " josephiNe-mc-vitty ", " sa-la-demOch "), firstAndLastNameFailedMsg("  mC  ", "  La  ", " josephiNe-mc-vitty ", " sa-la-demOch "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("  josephine MC ", " Sa la  ", " josephIne-mc-vitty  ", "  sa-la-deMoch "), firstAndLastNameFailedMsg("  josephine MC ", " Sa la  ", " josephIne-mc-vitty  ", "  sa-la-deMoch "));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" josephinE  ", "  sA ", "  josepHine-mc-vitty ", " sa-LA-democh  "), firstAndLastNameFailedMsg(" josephinE  ", "  sA ", "  josepHine-mc-vitty ", " sa-LA-democh  "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" josephine viTty ", " sa dEmoch ", " josePhine-mc-vitty ", "  Sa-la-democh  "), firstAndLastNameFailedMsg(" josephine viTty ", " sa dEmoch ", " josePhine-mc-vitty ", "  Sa-la-democh  "));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" JosePh  ", "  leE", " JosEphine-Mc-Vitty", " sa-la-Cathleen "), firstAndLastNameFailedMsg(" JosePh  ", "  leE", " JosEphine-Mc-Vitty", " sa-la-Cathleen "));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("  JOsephine-Mc-Vitty ", "sa-la-cathLeen  ", "joSeph ", " la lEe "), firstAndLastNameFailedMsg("  JOsephine-Mc-Vitty ", "sa-la-cathLeen  ", "joSeph ", " la lEe "));
+    }
+
+    //----------------------------------------------------------------------- Last and First Names Swapped ----------------------------------------------------------------------
+    @Test
+    void doNamesMatch_Maybe_FirstLastSwap_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" joHn  ", "  Doe ", " doe", "JOhn "), firstAndLastNameFailedMsg(" joHn  ", "  Doe ", " doe", "JOhn "));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_FirstLastSwap_CaseInsensitive_IgnoreSpaces_Punctuation() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" aN-Day-denIse  ", "  Sa-LA-Democh ", " sa le democh", "An dE Denise "), firstAndLastNameFailedMsg(" aN-Day-denIse  ", "  Sa-LA-Democh ", " sa le democh", "An dE Denise "));
+    }
+
+    @Test
+    void doNamesMatch_Maybe_FirstLastSwap_CaseInsensitive_IgnoreSpaces_Hyphenated() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" Josephine-mc-vitty", "Sa-la-democh ", "democH ", " vittY"), firstAndLastNameFailedMsg(" Josephine-mc-vitty", "Sa-la-democh ", "democH ", " vittY"));
```

Comment:
Are these reallly needed? We already have tests for names containing hyphens and tests for the swapping of first and last names. Do this tests provide coverage that isn't already handled in earlier tests?

Reply:
If I have to write in identifiers, I want to ensure the order logic is correct. I think this would help with that.